### PR TITLE
Create machineconfig for m5d

### PR DESCRIPTION
**Pull request by hongkailiu**
Ref [1] and [2].

[1].  https://docs.google.com/document/d/1G5Bw2J1vz8ff6n3Z6LCQmpReC3tHYTvMrUF9YgCClH4/edit#

[2]. https://github.com/cgwalters/playground/blob/d72d980b66cc2cb84e658140e5a7406691457aad/ignition/rhcos-raid-containers.json

Will remove `_` prefix of the folder after review

### CODE REVIEW

File: clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  config:
+    ignition:
+      version: 2.2.0
+    storage:
+      disks:
+      - device: "/dev/nvme1n1"
+        wipeTable: true
+        partitions:
+        - label: CONTR
+          sizeMiB: 0
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml:17]: Copied from example.
Should we give some real number?

File: clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: m5d
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml:6]: A new one like this, or modify an existing one?

File: clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+spec:
+  config:
+    ignition:
+      version: 2.2.0
+    storage:
+      disks:
+      - device: "/dev/nvme1n1"
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml:13]: raid0 with "/dev/nvme2n1"?

**Comment by alvaroaleman** [comment]
@hongkailiu can you just try out if it works by manually applying (and figure out first how to make the resulting nodes tainted so no test workloads end up on it)?

**Comment by hongkailiu** [comment]
> @hongkailiu can you just try out if it works by manually applying (and figure out first how to make the resulting nodes tainted so no test workloads end up on it)?

I will go through the review before i apply to it.
No idea what the effect will be.

File: clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml (hunk 2 above)
Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml:6]: What the effects to the current nodes?

File: clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        enable: true
+        contents: |-
+          [Mount]
+          What=/dev/md/containerraid
+          Where=/var/lib/containers
+          Type=xfs
+
+          [Install]
+          WantedBy=local-fs.target
+      - dropins:
```
Comment (cgwalters) [inline clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml:50]: You can drop the rest of the stuff here and below, this was just helpful for me testing locally to autologin on the console.

File: clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml (hunk 2 above)
Comment (cgwalters) [reply clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml:6]: It should be fine to make a new machineset like this.  You can then perform a rollout by scaling down the old machineset and scaling up this one.

File: clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml (hunk 4 above)
Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/_machine_config/m5d_machineconfig.yaml]: Done.

**Comment by cgwalters** [comment]
> (and figure out first how to make the resulting nodes tainted so no test workloads end up on it)?

If the Ignition config fails, the nodes won't join the cluster at all.  (But, I guess it's possible to have the config silently do the wrong thing, so it is worth doing with a machineset of size 1 indeed)

**Pull request merged**